### ISSUE TITLE
ICE when parsing `use`

### ISSUE THREAD

**Issue by Ekleog** [xd009642/tarpaulin#101]
I recently switched to `rustfmt-preview`, which grouped my `use` statements.

But this made `cargo tarpaulin --all` start to ICE:
```
error: expected one of `,` or `as`, found `::`
 --> /home/travis/build/Ekleog/yuubind/smtp-message/src/parse_helpers.rs:1:22
  |
1 | use std::{collections::HashMap, fmt};
  |                      ^^ expected one of `,` or `as` here
error: internal compiler error: Error constructed but not emitted
```

The same code works perfectly fine with both stable, beta and nightly vanilla rust.

You can also check [the error on Travis](https://travis-ci.org/Ekleog/yuubind/jobs/368599441#L603) and [rust stable managing to build](https://travis-ci.org/Ekleog/yuubind/jobs/368599438).

Anyway, thanks for your work!

**Comment by xd009642** [comment]
Huh, that's odd, checking it out now to see if I can get to the bottom. As it's ICE it actually crashes in rustc so it might be the rustflags and arguments I use to get the info/layout I need for coverage causes the crash.  I'd also raise an issue in rustc, the flags we add are `-C relocation-model=dynamic-no-pic -C link-dead-code`

**Comment by xd009642** [comment]
Unless it's down to syntex_synax... Which would be an issue.

**Comment by Ekleog** [comment]
Great, thanks!

**Comment by xd009642** [comment]
This is fixed with #88, it does require an addition to your RUSTFLAGS when tarpaulin builds so `proc-macro2` includes it's position reporting code which is semver-exempt. Looking in the develop branch and the new `.cargo/config` it should be clear what to do, but any issues getting it working let me know :+1:

**Comment by Ekleog** [comment]
Hmm… So I guess I'm mis-building tarpaulin? I'm building [this way](https://github.com/Ekleog/yuubind/blob/test-travis/.travis.yml#L22), got [apparent success](https://travis-ci.org/Ekleog/yuubind/jobs/377488103) (despite the test failure it appeared to successfully upload to codecov), yet get [no coverage](https://codecov.io/gh/Ekleog/yuubind)?

At worst I can just wait for 0.6.0, anyway waiting 20 minutes for every travis build is not really pleasant :)

Anyway, thanks! :)
